presto-main/src/test/java/com/facebook/presto/server/TestServer.java

```diff
@@ -135,6 +147,63 @@ public void testServerStarts()
         assertEquals(response.getStatusCode(), OK.getStatusCode());
     }
 
+    @Test
+    public void testBinaryResults()
+    {
+        // start query
+        URI uri = HttpUriBuilder.uriBuilderFrom(server.getBaseUrl())
+                .replacePath("/v1/statement")
+                .replaceParameter("binaryResults", "true")
+                .build();
+        Request request = preparePost()
+                .setUri(uri)
+                .setBodyGenerator(createStaticBodyGenerator("show catalogs", UTF_8))
+                .setHeader(PRESTO_USER, "user")
+                .setHeader(PRESTO_SOURCE, "source")
+                .setHeader(PRESTO_CATALOG, "catalog")
+                .setHeader(PRESTO_SCHEMA, "schema")
+                .setHeader(PRESTO_CLIENT_INFO, "{\"clientVersion\":\"testVersion\"}")
+                .build();
+
+        QueryResults queryResults = client.execute(request, createJsonResponseHandler(QUERY_RESULTS_CODEC));
+
+        ImmutableList.Builder<Object> data = ImmutableList.builder();
+        while (queryResults.getNextUri() != null) {
+            Request nextRequest = prepareGet()
+                    .setUri(queryResults.getNextUri())
+                    .build();
+            queryResults = client.execute(nextRequest, createJsonResponseHandler(QUERY_RESULTS_CODEC));
+
+            assertNull(queryResults.getData());
+            if (queryResults.getBinaryData() != null) {
+                data.addAll(queryResults.getBinaryData());
+            }
+        }
+
+        if (queryResults.getError() != null) {
```

Comment:
nit: Is there a Util class where we can create this helper method that is an improvement on assertNull(queryResults.getError()) that is used in a couple of places now (this whole check and fail descriptively thing) :-)